### DISCUSSION TITLE
Any solution for fetching static images dynamicly ?

### DISCUSSION THREAD

**Discussion by mayumih387** [gatsbyjs/gatsby#35092] [Help] [answered]
Hello,

I understand StaticImage doesn't accept dynamic src props to show images.

But what I wanted to do in some archive page is:
```
<StaticImage src={`../images/featuredImage/${node.slug}.jpg`} alt="" />
```

- blog post itself: headless CMS (1000+ posts)
- each featured image: /src/images/featuredImage/{slug}.jpg
A blog post "slug" is the same as the "name" of featured image (using gatsby-source-filesystem)

Any good idea to make it while still keeping featured images locally (not on a CMS) ?
Thank you for your help !

**Comment by LekoArts** [comment, marked as answer]
You'll then want to query for your image in the page and use `GatsbyImage` component: https://www.gatsbyjs.com/docs/how-to/images-and-media/using-gatsby-plugin-image/#dynamic-images

You can use https://www.gatsbyjs.com/plugins/gatsby-source-filesystem/ to get the images into GraphQL and then filter by `slug` on the name of the file

**Reply by mayumih387** [reply]
Thank you for your reply.

But unfortunately I'm still struggling with using `GatsbyImage` + dynamic image props inside a loop (map function).

```
{data.myApi.entries.map((node) => (
<li key={node.id}>
  <Link to={node.slug}>
    <GatsbyImage image={data.file.????} alt="" />
    // I want to call a local image with node.slug; each image can be different according to its post slug.
    <h2>{node.title}</h2>
 </Link>
</li>
))}
```

The "entries" query doesn't have featured image prop, but each featured images are stored in src/images/featuredImage with the same name with entry slug.

**Reply by LekoArts** [reply]
For this I'd create a foreign key relationship then, where the slug gets you the `File` node that you then can use `childImageSharp` on: https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#foreign-key-fields

**Reply by mayumih387** [reply]
Thank you for providing me a great idea.
Though it took me time to understand what it was, I've reached out.

Finally, I choose createResolvers API instead of using a foreign key in order to create "featuredImage" schema.

gatsby-node.js
```
exports.createResolvers = ({ createResolvers }) => {
  const resolvers = {
    myApi: {
      featuredImage: {
        type: "File",
        resolve(source, context) {
          return context.nodeModel.findOne({
            query: {
              filter: { name: { eq: source.slug } }
            },
            type: "File",
          })
        },
      },
    },
  }
  createResolvers(resolvers)
}
```

**Reply by LekoArts** [reply]
That also works 👍
